File: Sorting/JavaScript/bubbleSortBasic.js
// sort the following array using bubble sort

const arr = [11, 23, 8, 7, 10, 16, 14, 15];

console.log("unsorted Array: ", arr)
let swapped;

for (let i = 0; i < arr.length - 1; i++) {
  swapped = false;
  for (let j = 0; j < arr.length - 1 - i; j++) {
    if (arr[j] > arr[j + 1]) {
      const temp = arr[j];
      arr[j] = arr[j + 1];
      arr[j + 1] = temp;
      swapped = true;
    }
  }
  // if no value is swapped in the above loop it means that our array is sorted and there is no reason that we should go ahead with other iterations in the loop
  if (!swapped) {
    break;
  }
}

console.log("sorted array: ", arr)

File: Sorting/src/com/vinayak/BubbleSortBasic.java
package com.vinayak;

import java.util.Arrays;

public class BubbleSortBasic {
    public static void main(String[] args) {
        // sort the given array using bubble sort
        int[] arr = {1, 2, 3, 4, 5};
        boolean swapped;
        for (int i = 0; i < arr.length - 1; i++) {
            swapped = false;
            for (int j = 0; j < arr.length - 1 - i ; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }

            if (!swapped) {
                break;
            }
        }
        System.out.println("Sorted array: " + Arrays.toString(arr));
    }


}
